`first` commands stopped working when the key is `Hash`

### Bug description
Since Hash class refactoring, `first` commands do not work when the key is `Hash`.
this is due to `index.newInstance()` fails

### Steps To Reproduce
```java
    @Test
    public void testTangleFirstHash() throws Exception {

        RocksDBPersistenceProvider localRocksDBPP = new RocksDBPersistenceProvider("mainnetdb",
                "mainnetdb.log",1000);
        Tangle localTangle = new Tangle();
        localTangle.addPersistenceProvider(localRocksDBPP);
        localTangle.init();

        TransactionViewModel tx = TransactionViewModel.first(localTangle);
    }
```
### Expected behaviour
The first transaction in a database is loaded.

### Actual behaviour
an Exception is thrown.

### Errors
```
java.lang.InstantiationException: com.iota.iri.model.Hash

	at java.lang.Class.newInstance(Class.java:427)
	at com.iota.iri.storage.rocksDB.RocksDBPersistenceProvider.modelAndIndex(RocksDBPersistenceProvider.java:245)
	at com.iota.iri.storage.rocksDB.RocksDBPersistenceProvider.first(RocksDBPersistenceProvider.java:290)
	at com.iota.iri.storage.Tangle.getFirst(Tangle.java:188)
	at com.iota.iri.controllers.TransactionViewModel.first(TransactionViewModel.java:189)
	at com.iota.iri.storage.TangleTest.profileDBSectionsSize(TangleTest.java:80)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.lang.NoSuchMethodException: com.iota.iri.model.Hash.<init>()
	at java.lang.Class.getConstructor0(Class.java:3082)
	at java.lang.Class.newInstance(Class.java:412)
	... 29 more
```

--- a/src/main/java/com/iota/iri/controllers/AddressViewModel.java
+++ b/src/main/java/com/iota/iri/controllers/AddressViewModel.java
@@ -1,5 +1,6 @@
 package com.iota.iri.controllers;
 
+import com.iota.iri.model.AddressHash;
 import com.iota.iri.model.Hash;
 import com.iota.iri.model.persistables.Address;
 import com.iota.iri.storage.Indexable;
@@ -59,7 +60,7 @@ public class AddressViewModel implements HashesViewModel {
      * @throws Exception Thrown if the database fails to return a first object
      */
     public static AddressViewModel first(Tangle tangle) throws Exception {
-        Pair<Indexable, Persistable> bundlePair = tangle.getFirst(Address.class, Hash.class);
+        Pair<Indexable, Persistable> bundlePair = tangle.getFirst(Address.class, AddressHash.class);
         if(bundlePair != null && bundlePair.hi != null) {
             return new AddressViewModel((Address) bundlePair.hi, (Hash) bundlePair.low);
         }

--- a/src/main/java/com/iota/iri/controllers/ApproveeViewModel.java
+++ b/src/main/java/com/iota/iri/controllers/ApproveeViewModel.java
@@ -1,14 +1,13 @@
 package com.iota.iri.controllers;
 
 import com.iota.iri.model.Hash;
+import com.iota.iri.model.TransactionHash;
 import com.iota.iri.model.persistables.Approvee;
 import com.iota.iri.storage.Indexable;
 import com.iota.iri.storage.Persistable;
 import com.iota.iri.storage.Tangle;
 import com.iota.iri.utils.Pair;
 
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Set;
 
 /**
@@ -61,7 +60,7 @@ public class ApproveeViewModel implements HashesViewModel {
      * @throws Exception Thrown if the database fails to return a first object
      */
     public static ApproveeViewModel first(Tangle tangle) throws Exception {
-        Pair<Indexable, Persistable> bundlePair = tangle.getFirst(Approvee.class, Hash.class);
+        Pair<Indexable, Persistable> bundlePair = tangle.getFirst(Approvee.class, TransactionHash.class);
         if(bundlePair != null && bundlePair.hi != null) {
             return new ApproveeViewModel((Approvee) bundlePair.hi, (Hash) bundlePair.low);
         }

--- a/src/main/java/com/iota/iri/controllers/BundleViewModel.java
+++ b/src/main/java/com/iota/iri/controllers/BundleViewModel.java
@@ -1,5 +1,6 @@
 package com.iota.iri.controllers;
 
+import com.iota.iri.model.BundleHash;
 import com.iota.iri.model.Hash;
 import com.iota.iri.model.persistables.Bundle;
 import com.iota.iri.storage.Indexable;
@@ -7,8 +8,6 @@ import com.iota.iri.storage.Persistable;
 import com.iota.iri.storage.Tangle;
 import com.iota.iri.utils.Pair;
 
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Set;
 
 /**
@@ -61,7 +60,7 @@ public class BundleViewModel implements HashesViewModel {
      * @throws Exception Thrown if the database fails to return a first object
      */
     public static BundleViewModel first(Tangle tangle) throws Exception {
-        Pair<Indexable, Persistable> bundlePair = tangle.getFirst(Bundle.class, Hash.class);
+        Pair<Indexable, Persistable> bundlePair = tangle.getFirst(Bundle.class, BundleHash.class);
         if(bundlePair != null && bundlePair.hi != null) {
             return new BundleViewModel((Bundle) bundlePair.hi, (Hash) bundlePair.low);
         }

--- a/src/main/java/com/iota/iri/controllers/TagViewModel.java
+++ b/src/main/java/com/iota/iri/controllers/TagViewModel.java
@@ -1,6 +1,7 @@
 package com.iota.iri.controllers;
 
 import com.iota.iri.model.Hash;
+import com.iota.iri.model.TagHash;
 import com.iota.iri.model.persistables.ObsoleteTag;
 import com.iota.iri.model.persistables.Tag;
 import com.iota.iri.storage.Indexable;
@@ -8,8 +9,6 @@ import com.iota.iri.storage.Persistable;
 import com.iota.iri.storage.Tangle;
 import com.iota.iri.utils.Pair;
 
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Set;
 
 /**
@@ -94,7 +93,7 @@ public class TagViewModel implements HashesViewModel {
      * @throws Exception Thrown if the database fails to return a first object
      */
     public static TagViewModel first(Tangle tangle) throws Exception {
-        Pair<Indexable, Persistable> tagPair = tangle.getFirst(Tag.class, Hash.class);
+        Pair<Indexable, Persistable> tagPair = tangle.getFirst(Tag.class, TagHash.class);
         if(tagPair != null && tagPair.hi != null) {
             return new TagViewModel((Tag) tagPair.hi, (Hash) tagPair.low);
         }

--- a/src/main/java/com/iota/iri/controllers/TransactionViewModel.java
+++ b/src/main/java/com/iota/iri/controllers/TransactionViewModel.java
@@ -1,13 +1,8 @@
 package com.iota.iri.controllers;
 
 import com.iota.iri.model.*;
+import com.iota.iri.model.persistables.*;
 import com.iota.iri.service.snapshot.Snapshot;
-import com.iota.iri.model.persistables.Address;
-import com.iota.iri.model.persistables.Approvee;
-import com.iota.iri.model.persistables.Bundle;
-import com.iota.iri.model.persistables.ObsoleteTag;
-import com.iota.iri.model.persistables.Tag;
-import com.iota.iri.model.persistables.Transaction;
 import com.iota.iri.storage.Indexable;
 import com.iota.iri.storage.Persistable;
 import com.iota.iri.storage.Tangle;
@@ -246,7 +241,7 @@ public class TransactionViewModel {
      * @throws Exception Thrown if the database fails to return a first object.
      */
     public static TransactionViewModel first(Tangle tangle) throws Exception {
-        Pair<Indexable, Persistable> transactionPair = tangle.getFirst(Transaction.class, Hash.class);
+        Pair<Indexable, Persistable> transactionPair = tangle.getFirst(Transaction.class, TransactionHash.class);
         if (transactionPair != null && transactionPair.hi != null) {
             return new TransactionViewModel((Transaction) transactionPair.hi, (Hash) transactionPair.low);
         }

--- a/src/main/java/com/iota/iri/model/AbstractHash.java
+++ b/src/main/java/com/iota/iri/model/AbstractHash.java
@@ -1,9 +1,5 @@
 package com.iota.iri.model;
 
-import java.io.Serializable;
-import java.util.Arrays;
-import java.util.Objects;
-
 import com.iota.iri.crypto.Curl;
 import com.iota.iri.model.persistables.Transaction;
 import com.iota.iri.model.safe.ByteSafe;
@@ -11,11 +7,18 @@ import com.iota.iri.model.safe.TritSafe;
 import com.iota.iri.storage.Indexable;
 import com.iota.iri.utils.Converter;
 
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Objects;
+
 public abstract class AbstractHash implements Hash, Serializable {
     private final Object lock = new Object();
 
     private ByteSafe byteSafe;
     private TritSafe tritSafe;
+
+    public AbstractHash() {
+    }
 
     public AbstractHash(byte[] source, int sourceOffset, int sourceSize) {
         if(sourceSize < SIZE_IN_TRITS) {

--- a/src/main/java/com/iota/iri/model/AddressHash.java
+++ b/src/main/java/com/iota/iri/model/AddressHash.java
@@ -2,6 +2,8 @@ package com.iota.iri.model;
 
 public class AddressHash extends AbstractHash {
 
+    public AddressHash() { }
+
     protected AddressHash(byte[] bytes, int offset, int sizeInBytes) {
         super(bytes, offset, sizeInBytes);
     }

--- a/src/main/java/com/iota/iri/model/BundleHash.java
+++ b/src/main/java/com/iota/iri/model/BundleHash.java
@@ -2,6 +2,8 @@ package com.iota.iri.model;
 
 public class BundleHash extends AbstractHash {
 
+    public BundleHash() { }
+
     protected BundleHash(byte[] bytes, int offset, int sizeInBytes) {
         super(bytes, offset, sizeInBytes);
     }

--- a/src/main/java/com/iota/iri/model/TagHash.java
+++ b/src/main/java/com/iota/iri/model/TagHash.java
@@ -2,6 +2,8 @@ package com.iota.iri.model;
 
 public class TagHash extends AbstractHash {
 
+    public TagHash() { }
+
     protected TagHash(byte[] tagBytes, int offset, int tagSizeInBytes) {
         super(tagBytes, offset, tagSizeInBytes);
     }

--- a/src/main/java/com/iota/iri/model/TransactionHash.java
+++ b/src/main/java/com/iota/iri/model/TransactionHash.java
@@ -6,7 +6,9 @@ import com.iota.iri.utils.Converter;
 
 public class TransactionHash extends AbstractHash {
 
-	protected TransactionHash(byte[] source, int offset, int sourceSize) {
+    public TransactionHash() { }
+
+    protected TransactionHash(byte[] source, int offset, int sourceSize) {
 		super(source, offset, sourceSize);
 	}
 	

--- a/src/test/java/com/iota/iri/controllers/BundleViewModelTest.java
+++ b/src/test/java/com/iota/iri/controllers/BundleViewModelTest.java
@@ -1,8 +1,14 @@
 package com.iota.iri.controllers;
 
+import com.iota.iri.conf.MainnetConfig;
+import com.iota.iri.crypto.SpongeFactory;
+import com.iota.iri.model.TransactionHash;
+import com.iota.iri.service.snapshot.SnapshotProvider;
+import com.iota.iri.service.snapshot.impl.SnapshotProviderImpl;
 import com.iota.iri.storage.Tangle;
 import com.iota.iri.storage.rocksDB.RocksDBPersistenceProvider;
 import org.junit.After;
+import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
@@ -14,6 +20,7 @@ public class BundleViewModelTest {
     private static final TemporaryFolder dbFolder = new TemporaryFolder();
     private static final TemporaryFolder logFolder = new TemporaryFolder();
     private static Tangle tangle = new Tangle();
+    private static SnapshotProvider snapshotProvider;
 
     @Before
     public void setUp() throws Exception {
@@ -24,6 +31,7 @@ public class BundleViewModelTest {
                 logFolder.getRoot().getAbsolutePath(),1000);
         tangle.addPersistenceProvider(rocksDBPersistenceProvider);
         tangle.init();
+        snapshotProvider = new SnapshotProviderImpl().init(new MainnetConfig());
 
     }
 
@@ -32,6 +40,8 @@ public class BundleViewModelTest {
         tangle.shutdown();
         dbFolder.delete();
         logFolder.delete();
+        snapshotProvider.shutdown();
+
     }
 
     @Test
@@ -57,6 +67,16 @@ public class BundleViewModelTest {
     @Test
     public void getTail() throws Exception {
 
+    }
+
+    @Test
+    public void firstShouldFindTx() throws Exception {
+        byte[] trits = TransactionViewModelTest.getRandomTransactionTrits();
+        TransactionViewModel transactionViewModel = new TransactionViewModel(trits, TransactionHash.calculate(SpongeFactory.Mode.CURLP81, trits));
+        transactionViewModel.store(tangle, snapshotProvider.getInitialSnapshot());
+
+        BundleViewModel result = BundleViewModel.first(tangle);
+        Assert.assertTrue(result.getHashes().contains(transactionViewModel.getHash()));
     }
 
 }

--- a/src/test/java/com/iota/iri/controllers/TransactionViewModelTest.java
+++ b/src/test/java/com/iota/iri/controllers/TransactionViewModelTest.java
@@ -402,6 +402,16 @@ public class TransactionViewModelTest {
         log.info("Done. #TX: {}", TransactionViewModel.getNumberOfStoredTransactions(tangle));
     }
 
+    @Test
+    public void firstShouldFindTx() throws Exception {
+        byte[] trits = getRandomTransactionTrits();
+        TransactionViewModel transactionViewModel = new TransactionViewModel(trits, TransactionHash.calculate(SpongeFactory.Mode.CURLP81, trits));
+        transactionViewModel.store(tangle, snapshotProvider.getInitialSnapshot());
+
+        TransactionViewModel result = TransactionViewModel.first(tangle);
+        Assert.assertEquals(transactionViewModel.getHash(), result.getHash());
+    }
+
     private Transaction getRandomTransaction(Random seed) {
         Transaction transaction = new Transaction();
 

--- a/src/test/java/com/iota/iri/controllers/TransactionViewModelTest.java
+++ b/src/test/java/com/iota/iri/controllers/TransactionViewModelTest.java
@@ -11,9 +11,9 @@ import com.iota.iri.service.snapshot.impl.SnapshotProviderImpl;
 import com.iota.iri.storage.Tangle;
 import com.iota.iri.storage.rocksDB.RocksDBPersistenceProvider;
 import com.iota.iri.utils.Converter;
-import org.junit.AfterClass;
+import org.junit.After;
 import org.junit.Assert;
-import org.junit.BeforeClass;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 import org.slf4j.Logger;
@@ -36,8 +36,8 @@ public class TransactionViewModelTest {
 
     private static final Random seed = new Random();
 
-    @BeforeClass
-    public static void setUp() throws Exception {
+    @Before
+    public void setUp() throws Exception {
         dbFolder.create();
         logFolder.create();
         RocksDBPersistenceProvider rocksDBPersistenceProvider;
@@ -48,8 +48,8 @@ public class TransactionViewModelTest {
         snapshotProvider = new SnapshotProviderImpl().init(new MainnetConfig());
     }
 
-    @AfterClass
-    public static void tearDown() throws Exception {
+    @After
+    public void tearDown() throws Exception {
         tangle.shutdown();
         snapshotProvider.shutdown();
         dbFolder.delete();

